src/NServiceBus.Core/ServicePlatform/Retries/ManualRetryNotificationBehavior.cs

```diff
@@ -0,0 +1,60 @@
+﻿namespace NServiceBus
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Threading.Tasks;
+    using Pipeline;
+    using Routing;
+    using Transport;
+
+    class ManualRetryNotificationBehavior : IForkConnector<IIncomingLogicalMessageContext, IIncomingLogicalMessageContext, IRoutingContext>
+    {
+        const string RetryUniqueMessageIdHeader = "ServiceControl.Retry.UniqueMessageId";
+
+        readonly string errorQueue;
+
+        public ManualRetryNotificationBehavior(string errorQueue)
+        {
+            this.errorQueue = errorQueue;
+        }
+
+        public async Task Invoke(IIncomingLogicalMessageContext context, Func<IIncomingLogicalMessageContext, Task> next)
+        {
+            await next(context).ConfigureAwait(false);
+
+            if (IsRetriedMessage(out var id))
+            {
+                await ConfirmSuccessfulRetry().ConfigureAwait(false);
+            }
+
+            async Task ConfirmSuccessfulRetry()
+            {
+                var messageToDispatch = new OutgoingMessage(
+                    CombGuid.Generate().ToString(),
+                    new Dictionary<string, string>
+                    {
+                        { "ServiceControl.Retry.Successful", DateTimeOffset.UtcNow.ToString("O") },
+                        { RetryUniqueMessageIdHeader, id }
+                    },
+                    new byte[0]);
+                var routingContext = new RoutingContext(messageToDispatch, new UnicastRoutingStrategy(errorQueue), context);
+                await this.Fork(routingContext).ConfigureAwait(false);
```

Comment:
Is this message going to be captured by the audits when enabled?

Reply:
no, only the incoming message is going to be audited.